server.go

```diff
@@ -469,6 +464,28 @@ func (s *server) handleQuery(querymsg interface{}, state *peerState) {
 			msg.reply <- errors.New("peer not found")
 		}
 
+	case dropNodeMsg:
+		// Check inbound peers. We pass an empty callback since we don't
+		// require any additional actions on disconnect for inbound peers.
+		found := disconnectPeer(state.peers, msg.addr, func(p *peer) {})
```

Comment:
I'd suggest passing nil for the function here and adding a nil check in the `disconnectPeer` function.  This will avoid creating a closure which is much more expensive than a nil check.